.github/workflows/wheels-build.yaml

```diff
@@ -91,13 +91,19 @@ jobs:
           - { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.10',  LINUX_VER: 'centos7' }
           - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.9', LINUX_VER: 'rockylinux8' }
           - { CUDA_VER: '12.0.1', ARCH: 'arm64', PY_VER: '3.10', LINUX_VER: 'rockylinux8' }
+          - { CUDA_VER: '12.2.2', ARCH: 'amd64', PY_VER: '3.9',  LINUX_VER: 'centos7' }
+          - { CUDA_VER: '12.2.2', ARCH: 'amd64', PY_VER: '3.10',  LINUX_VER: 'centos7' }
+          - { CUDA_VER: '12.2.2', ARCH: 'arm64', PY_VER: '3.9', LINUX_VER: 'rockylinux8' }
+          - { CUDA_VER: '12.2.2', ARCH: 'arm64', PY_VER: '3.10', LINUX_VER: 'rockylinux8' }
```

Comment:
We only need to _build_ with one CUDA version from each major version. This is because of CUDA Enhanced Compatibility (CEC), which ensures that builds for CUDA 12.x are backwards compatible with CUDA 12.y at runtime (`y < x`). https://docs.nvidia.com/deploy/cuda-compatibility/

(Also, the wheels are labeled by CUDA major version like `-cu12`, not major.minor, so we wouldn't even be able to distinguish the 12.0 and 12.2 builds. This is intentional, because the guarantees we get from CEC mean that we don't need CUDA minor versions in the wheel names.)

We can drop builds with 12.0.1 but I would vote to retain (some? all?) tests for 12.0.1. In the past we've expressed the intention to test according to this policy (previously discussed with @raydouglass @ajschmidt8, but this policy may not be documented?):

- latest CUDA from the previous major version (11.8.0)
- earliest CUDA from the current major version (12.0.1)
- latest CUDA from the current major version (now 12.2.2) (* see below!)

Until this PR added 12.2, the last two bullets were the same (we built with 12.0 and tested with 12.0 so testing CEC was irrelevant since the build/test matrices matched). Adding 12.2 means that we want to add a test config for 12.2 while retaining 12.0 for CEC, to align with this policy.

The asterisk on the last one is that RAPIDS typically lags CUDA minor releases by a bit. For example: we could test with 12.3 in theory, but we don't have GPU CI runners with new enough drivers yet. We are sticking to 12.2 for now because it's the latest CUDA minor version that has a corresponding LTS driver branch (R535). https://docs.nvidia.com/datacenter/tesla/drivers/index.html#cuda-drivers

Reply:
I would vote to omit `12.0.1` from the PR test matrix for now so that we can keep the number of combinations to three.

Once we get on the new GPU cluster, we'll have more capacity and can expand the matrix. Until then, we can just increase the nightly matrix.

I worry that adding too many combinations to the PR tests will increase our queue times too much given the number of repositories that now use our shared workflows.

Reply:
> we can keep the number of combinations to three

That's fine with me (you know about our available runner capacity a lot better than me), but how about we accomplish that by instead having 1 CUDA 12.0 job and 1 CUDA 12.2 job for the PR test matrix?

With a mix of Python versions and architecture, like this:

```text
- { CUDA_VER: '12.0.1', ARCH: 'amd64', PY_VER: '3.9',  LINUX_VER: 'rockylinux8' }
- { CUDA_VER: '12.2.2', ARCH: 'arm64', PY_VER: '3.10',  LINUX_VER: 'rockylinux8' }
```

That'd keep the absolute number of jobs the same, but give us some coverage of both 12.0 and 12.2 on PRs, so that issues with relying on CUDA Enhanced Compatibility could be caught ealier.

Reply:
> I think we've concluded in the past with Vyas that it's better to test across CUDA versions than Python versions in PRs

Perfect, yeah I was thinking along those lines because of what he wrote at https://github.com/rapidsai/build-planning/issues/5.

> So then we'll drop one of the 11.8.0 jobs?

No, there's already only 1 11.8.0 job being run on PRs ([code link](https://github.com/rapidsai/shared-workflows/blob/91799a905608f4c57d7fe65b92ce9a261249e635/.github/workflows/wheels-test.yaml#L68-L71))

```text
- { ARCH: 'amd64', PY_VER: '3.9', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
- { ARCH: 'amd64', PY_VER: '3.10', CUDA_VER: '12.0.1', LINUX_VER: 'ubuntu20.04', gpu: 'v100', driver: 'latest' }
- { ARCH: 'arm64', PY_VER: '3.9', CUDA_VER: '12.0.1', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
```

And 3 total jobs.

I'm proposing preserving that 11.8.0 job, removing one of the 12.0.1 jobs, and adding one 12.2.0 job, like this:

```text
- { ARCH: 'amd64', PY_VER: '3.9', CUDA_VER: '11.8.0', LINUX_VER: 'ubuntu18.04', gpu: 'v100', driver: 'latest' }
- { ARCH: 'amd64', PY_VER: '3.9', CUDA_VER: '12.0.1', LINUX_VER: 'ubuntu20.04', gpu: 'v100', driver: 'latest' }
- { ARCH: 'arm64', PY_VER: '3.10', CUDA_VER: '12.2.2', LINUX_VER: 'ubuntu20.04', gpu: 'a100', driver: 'latest' }
```

Then CUDA 11.8.0, 12.0.1, and 12.2.2 would be tested on every PR. Without any change in the absolute number of `wheels-test` jobs per commit on PRs.

Reply:
I just pushed that change in https://github.com/rapidsai/shared-workflows/pull/166/commits/b4f97a69e2db7839941406647a1d6524c8a568e5.

So as of the current state of this branch, the PR test matrix:

* has 3 total jobs (just like on `branch-24.02`)
* has 1 job for each of CUDA 11.8, 12.0, 12.2
* spreads architecture, Python version, and Ubuntu version over those

Reply:
Great. I like this. I proposed a small tweak here: https://github.com/rapidsai/shared-workflows/pull/166/files#r1445403457